### PR TITLE
[Shortcuts]: Fix crash in the wild

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/31419

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Edit your Preferences file so that it contains an empty string:
```
  // change this:
  "brave": {
    "accelerators": {
      "33000": [
        "Alt+ArrowLeft",
        "AltGr+ArrowLeft",
        "BrowserBack"
      ],

  // to this:
  "brave": {
    "accelerators": {
      "33000": [
        "",
        "AltGr+ArrowLeft",
        "BrowserBack"
      ],
```
2. Launch Brave
3. It should not crash.

### CODE REVIEW

File: components/commands/common/accelerator_parsing.cc [hunk 1]
```diff
@@ -192,6 +192,14 @@ ui::Accelerator FromCodesString(const std::string& value) {
   std::vector<std::string> parts = base::SplitString(
       value, "+", base::TRIM_WHITESPACE, base::SPLIT_WANT_ALL);
 
+  // Not sure why, but some clients are encountering empty accelerators. If we
+  // encounter one in the wild, just return an empty accelerator instead of
+  // crashing:
+  // https://github.com/brave/brave-browser/issues/31419
+  if (!parts.size()) {
+    return ui::Accelerator();
+  }
```
Comment (sangwoo108) [inline components/commands/common/accelerator_parsing.cc:201]: Oops, the call stack seems to be from Mac. Might be related to the routine to get default accelerators?

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: components/commands/common/accelerator_parsing.cc [hunk 2]
```diff
@@ -192,6 +192,14 @@ ui::Accelerator FromCodesString(const std::string& value) {
   std::vector<std::string> parts = base::SplitString(
       value, "+", base::TRIM_WHITESPACE, base::SPLIT_WANT_ALL);
 
+  // Not sure why, but some clients are encountering empty accelerators. If we
+  // encounter one in the wild, just return an empty accelerator instead of
+  // crashing:
+  // https://github.com/brave/brave-browser/issues/31419
+  if (!parts.size()) {
```
Comment (goodov) [inline components/commands/common/accelerator_parsing.cc:199]: nit: `if (parts.empty()) {`

**Review by goodov — APPROVED** [review]

File: components/commands/common/accelerator_parsing.cc (hunk 1 above)
Comment (fallaciousreasoning) [reply components/commands/common/accelerator_parsing.cc:201]: I wonder if it's due to removing a shortcut in the system UI? Either way, this should hopefully stop the crash :laughing:

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.55.100 Chromium: 115.0.5790.56 (Official Build) nightly (64-bit)
-- | --
Revision | f6a25cd5f3e75378d97cb00782d9b95f0c7722fa
OS | Windows 11 Version 22H2 (Build 22621.1848)
```

Using the STR/Cases outlined via https://github.com/brave/brave-core/pull/19137#issue-1786906959, reproduced the crash using `1.55.97 Chromium: 115.0.5790.56`. Once the change was made in the `Preference` file, Brave wouldn't open.

Verified that once Brave was updated to `1.55.100 Chromium: 115.0.5790.56`, the same profile launched/worked which was crashing earlier when using `1.55.97 Chromium: 115.0.5790.56`.

**Pull request merged**